### PR TITLE
Add hint on FF addon. Fixes #201.

### LINKED ISSUE #201: grunt watch:ff fails
Reading https://github.com/dreditor/dreditor/blob/1.x/CONTRIBUTING.md

> Requires the Firefox Add-on SDK, which should have been installed by the initial Setup already.

I'm not sure the **initial Setup** did it's proposed job.

```
Running "autoload:ff" task
Warning: Auto-loading release/firefox/dreditor.xpi failed: (4) Error: --2014-06-07 11:49:06--  http://localhost:8888/
Resolving localhost... ::1, 127.0.0.1, fe80::1
Connecting to localhost|::1|:8888... failed: Connection refused.
Connecting to localhost|127.0.0.1|:8888... failed: Connection refused.
Connecting to localhost|fe80::1|:8888... failed: Invalid argument.
```

Even running `brew install mozilla-addon-sdk` did not fix this.

What's wrong?

### PR DESCRIPTION
**Pull request by clemens-tolboom**
I never update FF so lets inform the user running `grunt ff:watch`

### CODE REVIEW

**Comment by markhalliwell** [comment]
``` sh
Gruntfile.js
  line 392  col 11  Bad line breaking before '+'.
  line 393  col 11  Bad line breaking before '+'.
✖ 2 problems
```

File: Gruntfile.js [hunk 1]
```diff
@@ -387,7 +387,10 @@ module.exports = function(grunt) {
     },
     function (error, result, code) {
       if (code !== 8) {
-        return grunt.warn('Auto-loading ' + xpi + ' failed: (' + code + ') ' + error);
+        var plugin = "https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/";
+        return grunt.warn('Auto-loading ' + xpi + ' failed: (' + code + ') ' +
+          error +
```
Comment (sun) [inline Gruntfile.js]: Let's keep error on the first line?

File: Gruntfile.js [hunk 2]
```diff
@@ -387,7 +387,10 @@ module.exports = function(grunt) {
...
+          "\n\nAre you sure you have installed the Firefox addon:\n- " + plugin);
```
Comment (sun) [inline Gruntfile.js]: Slightly weird punctuation here... why the `-` after the newline?

Also, _"Are you sure..."_ should end in a question mark.  Alternatively we can say _"Ensure you have..."_

**Comment by clemens-tolboom** [comment]
Latest commit gives:

``` bash
$ grunt autoload:ff
Running "autoload:ff" task
Warning: Auto-loading release/firefox/dreditor.xpi failed: (3) Error: --2014-06-21 14:48:40--  http://localhost:8888/
BODY data file 'release/firefox/dreditor.xpi' missing: No such file or directory

Are you sure you have installed the Firefox addon:
- https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/

 Use --force to continue.
```

My intent was something like a summation. Maybe this is better?

> Are you sure you have:
> -  installed the Firefox addon https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/ ?
> - other message

**Comment by sun** [comment]
Hm.  How about placing the code and error on a separate line?

> Autoloading {...xpi} failed:
> {code}: {error}
> 
> [...hint...]

That said, the hint in the last example error message is misleading, since the .xpi file does not exist. :-/

I'd go with the shortest and simplest hint:

> Ensure you have the AutoInstaller extension installed in Firefox:
> https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/

FWIW, I only ever run `grunt watch:ff` myself (except when running tests).

**Comment by clemens-tolboom** [comment]
I only wished I had any hint.

File: Gruntfile.js [hunk 3]
```diff
@@ -387,7 +387,10 @@ module.exports = function(grunt) {
     },
     function (error, result, code) {
       if (code !== 8) {
-        return grunt.warn('Auto-loading ' + xpi + ' failed: (' + code + ') ' + error);
+        var hint = "Ensure you have the AutoInstaller extension installed in Firefox: ";
+        var plugin = "https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/";
+        return grunt.warn('Auto-loading ' + xpi + ' failed: (' + code + ') ' + error +
+          "\n\n" + hint + "\n- " + plugin + "\n\n");
```
Comment (markhalliwell) [inline Gruntfile.js]: Creating multiple variables for one message is unnecessary. Also the code/error really should be on it's own line:

``` js
return grunt.warn('Auto-loading ' + xpi + ' failed:\n\n' +
  code + ': ' + error + '\n\n' +
  'Ensure you have the AutoInstaller extension installed in Firefox:\n' +
  'https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/\n\n'
);
```

**Pull request merged**